### PR TITLE
Alerting: Fix Recording Rule QueryEditor builder view

### PR DESCRIPTION
**Pull request by VikaCep**
This PR fixes the Builder button in the query editor for Recording Rules as previously it was not performing any action upon clicking it.

**Who is this feature for?**

All users

Before:

After:

### CODE REVIEW

**Comment by konrad147** [comment]
Shouldn't we backport it to v10.1?

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/reducer.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export const queriesAndExpressionsReducer = createReducer(initialState, (builder
       const query = payload.recordingRuleQueries[0];
       const recordingRuleQuery = {
         ...query,
-        ...{ expr: payload.expression, model: { expr: payload.expression, refId: query.model.refId } },
+        ...{ expr: payload.expression, model: payload.recordingRuleQueries[0].model },
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/reducer.ts:83]: Are we sure the `recordingRuleQueries` array is not empty?

**Comment by gillesdemey** [comment]
Does this not need to be back-ported? Does it only occur in 10.2 and not 10.1?

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/reducer.ts (hunk 1 above)
Comment (gillesdemey) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/reducer.ts:83]: I think it's safest to use optional chaining here (and this has also been defined as `query` variable above so no need for the entire payload path here)

**Comment by VikaCep** [comment]
@konrad147 @gillesdemey you're right, backporting to v10.1.x

**Review by gillesdemey — APPROVED** [review]
Works as intended – LGTM! ✅

**Pull request merged**